receiver/prometheusreceiver/config.go

```diff
@@ -48,26 +46,6 @@ func (cfg *Config) Validate() error {
 	return nil
 }
 
-type TargetAllocator struct {
```

Comment:
This is technically an API breaking change, correct? We are removing the previously exported type `TargetAllocator` and some others? If that's the case, can we add a changelog item with `change_logs: ["api"]`.

Reply:
done